### PR TITLE
[SCVMM] Public Preview refresh using New Resource Model

### PR DESCRIPTION
**Pull request by nascarsayan**
* Using New Resource Model for all VM operations:
    * `vm` command : Create, Update, Delete, Show, VirtualMachineInstance and Perform Operations on VM Instance
    * `vm guest-agent` command : Enable, Show VMInstanceGuestAgent
    * `vm extension` command : Create, Update, Delete, Show, List MachineExtensions
* If underlying machine is not present, it gets created during vm create
* Added delete-from-host flag for `vm delete`
* Deprecated VM List option as VM Instance is a child resource of Machines.
* Updated tests and helps accordingly.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
SCVMM

**Comment by yanzhudd** [comment]
If you want to release a new version of the extension, please change the version in setup.py file.

File: src/scvmm/azext_scvmm/_params.py [hunk 1]
```diff
@@ -290,3 +318,77 @@ def load_arguments(self: AzCommandsLoader, _):
...
+    with self.argument_context('scvmm vm extension show') as c:
+        c.argument(
+            'vm_name', type=str, help='The name of the vm containing the extension.',
+            id_part='name')
+        c.argument('name', type=str, help='The name of the vm extension.', id_part='child_name_1')
+
+    for scope in ['scvmm vm extension update', 'scvmm vm extension create']:
+        with self.argument_context(scope) as c:
+            c.argument(
+                'vm_name', type=str, help='The name of the vm where the extension '
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_params.py:364]: It is not necessary to specify the `str` type. `str` is the default value for `type` field.
```suggestion
                'vm_name', help='The name of the vm where the extension '
```

Comment (nascarsayan) [reply src/scvmm/azext_scvmm/_params.py]: Removed type = str from all places in this file.

**Comment by nascarsayan** [comment]
> If you want to release a new version of the extension, please change the version in setup.py file.

We don't want to release this to the customers yet, we'll do that when around the time of Public Preview

File: src/scvmm/azext_scvmm/commands.py [hunk 2]
```diff
@@ -142,17 +165,20 @@ def load_command_table(self: AzCommandsLoader, _):
         )
         g.custom_command('update', 'update_vm', supports_no_wait=True)
         g.custom_show_command('show', 'show_vm')
-        g.custom_command('list', 'list_vm')
+        g.custom_command('list', 'list_vm',
+                         deprecate_info=g.deprecate(redirect='az scvmm vm show', hide=True))
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/commands.py:169]:
```suggestion
                         deprecate_info=g.deprecate(redirect='scvmm vm show', hide=True))
```

Comment (yanzhudd) [reply src/scvmm/azext_scvmm/commands.py:169]: May I ask the reason for this redirection?

Comment (nascarsayan) [reply src/scvmm/azext_scvmm/commands.py]: The VMInstance is a child resource of Machine resource type. We don't have a direct API support in ARM for listing all the child resources in a  subscription or a resourceGroup. We could use ARG queries to do that, but that's not yet finalised.
If required, we'll revisit this command later, after discussing internally.

**Comment by yanzhudd** [comment]
Please add tests for new added and changed commands.

File: src/scvmm/azext_scvmm/commands.py (hunk 2 above)
Comment (yanzhudd) [reply src/scvmm/azext_scvmm/commands.py]: Does the behavior of `list` command be changed, i.e., Was the listing feature not supported in the previous version?

**Comment by nascarsayan** [comment]
> Please add tests for new added and changed commands.

@yanzhudd , the tests for the new commandlets (Guest Agent and HCRP Extensions) will be added in a subsequent PR, by a core team member from VMM.
CC: @miniagarwal0910

**Review by yanzhudd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

File: src/scvmm/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,17 @@
 Release History
 ===============
 
+0.2.0
```
Comment (zhoxing-ms) [inline src/scvmm/HISTORY.rst:6]: If you don't want to release a new extension, please mark the version in `HISTORY.rst` as `Upcoming`

**Pull request merged**